File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        char arr[s.size()];
        int top = -1;  
        for (char ch : s) {
    
            if (ch == '(' || ch == '{' || ch == '[') {
                top++;  
                arr[top] = ch;  
            } 
            else {
                if (top < 0 || (ch == ')' && arr[top] != '(') || 
                              (ch == '}' && arr[top] != '{') || 
                              (ch == ']' && arr[top] != '[')) {
                    return false;
                }
                top--;  
            }
        } 
        return top == -1;
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) < 2:
            return False

        char_map = {'}':'{',']':'[',')':'('}
        stack = []

        for c in s:
            if c in char_map.values():
                stack.append(c)
            else:
                if stack and stack[-1] == char_map[c]:
                    stack.pop()  
                else:
                    return False
                    

        

        return len(stack) == 0

